apis/field_error.go

```diff
@@ -85,17 +92,57 @@ func (fe *FieldError) ViaFieldIndex(field string, index int) *FieldError {
 //    }
 //  }
 func (fe *FieldError) ViaKey(key string) *FieldError {
-	if fe == nil {
-		return nil
-	}
-	return fe.ViaField(fmt.Sprintf("[%s]", key))
+	return fe.ViaField(asKey(key))
 }
 
 // ViaFieldKey is the short way to chain: err.ViaKey(bar).ViaField(foo)
 func (fe *FieldError) ViaFieldKey(field string, key string) *FieldError {
 	return fe.ViaKey(key).ViaField(field)
 }
 
+// Also collects errors, returns a new collection of existing errors and new errors.
+func (fe *FieldError) Also(errs ...*FieldError) *FieldError {
+	newErr := &FieldError{}
+	// collect the current objects errors, if it has any
+	if fe != nil {
+		newErr.errors = fe.getNormalizedErrors()
+	}
+	// and then collect the passed in errors
+	for _, e := range errs {
+		newErr.errors = append(newErr.errors, e.getNormalizedErrors()...)
+	}
+	return newErr
+}
+
+func (fe *FieldError) getNormalizedErrors() []FieldError {
+	if fe == nil {
+		return []FieldError(nil)
```

Comment:
this cast feels superfluous, if intentional then add a comment explaining?

Reply:
Ah. this is guarding `(*FieldError).Error()` case